spp_event_demo/models/registrant.py

```diff
@@ -0,0 +1,30 @@
+# Part of OpenSPP. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class OpenSPPRegistrant(models.Model):
+    _inherit = "res.partner"
+
+    active_house_visit = fields.Many2one(
+        "spp.event.data", compute="_compute_active_house_visit"
+    )
+    active_phone_survey = fields.Many2one(
+        "spp.event.data", compute="_compute_active_phone_survey"
+    )
+
+    @api.depends("event_data_ids")
+    def _compute_active_house_visit(self):
+        for rec in self:
+            rec.active_house_visit = None
+            active_house_visit = rec._get_active_event("spp.event.house.visit")
+            if active_house_visit:
+                rec.active_house_visit = active_house_visit.id
```

Comment:
Instead of duplicating this code, can you add a function _get_active_event_id(...)
So you can simply do:
> rec.active_house_visit = rec._get_active_event_id("spp.event.house.visit")